src/app/pages/storage/components/dashboard-pool/export-disconnect-modal/export-disconnect-modal.component.ts

```diff
@@ -240,51 +267,116 @@ export class ExportDisconnectModalComponent implements OnInit {
     this.showUnknownStatusDetachWarning = this.pool.status === PoolStatus.Unknown;
-    this.showDestroy = this.pool.status !== PoolStatus.Unknown;
 
-    this.confirmLabelText = this.pool.status === PoolStatus.Unknown
-      ? (this.translate.instant(helptextVolumes.exportDialog.confirm)
-        + ' ' + this.translate.instant(helptextVolumes.exportDialog.unknownStatusAltText)) as TranslatedString
-      : this.translate.instant(helptextVolumes.exportDialog.confirm);
+    this.form.controls.destroy.valueChanges
+      .pipe(untilDestroyed(this))
+      .subscribe(() => this.resetNameInputValidState());
+  }
 
-    this.processes.forEach((process) => {
-      if (process.service) {
-        return;
-      }
+  selectOption(option: DisconnectOption | null): void {
+    this.selectedOption.set(option);
+    if (option) {
+      this.form.patchValue({
+        destroy: option === DisconnectOption.Delete,
+        confirm: false,
+      });
+      this.resetNameInputValidState();
+    }
+    this.cdr.markForCheck();
+  }
 
-      if (process.name && process.name !== '') {
-        this.process.knownProcesses.push(process);
-      } else {
-        this.process.unknownProcesses.push(process);
-      }
+  get isExportSelected(): boolean {
+    return this.selectedOption() === DisconnectOption.Export;
+  }
+
+  get isDeleteSelected(): boolean {
+    return this.selectedOption() === DisconnectOption.Delete;
+  }
+
+  get hasOtherPools(): boolean {
+    return this.otherPools.length > 0;
+  }
+
+  get shouldShowSystemDatasetWarning(): boolean {
+    return this.showSysDatasetWarning && this.isExportSelected;
+  }
+
+  get isDestinationPoolEncrypted(): boolean {
+    return this.destinationPool ? (this.poolEncryptionStatus[this.destinationPool.name] || false) : false;
+  }
+
+  canProceed(): boolean {
+    return this.selectedOption() !== null && this.form.valid && !this.isFormLoading;
+  }
+
+  private loadPoolEncryptionStatus(pools: Pool[]): void {
+    // Query the root dataset of each pool to determine if it's encrypted
+    const datasetQueries = pools.map((pool) => this.api.call('pool.dataset.query', [[['pool', '=', pool.name], ['type', '=', 'FILESYSTEM']], { extra: { retrieve_children: false } }]).pipe(
+      map((datasets: Dataset[]) => ({
+        poolName: pool.name,
+        encrypted: datasets.length > 0 && datasets[0].encrypted,
+      })),
+    ));
+
+    forkJoin(datasetQueries).pipe(
+      this.errorHandler.withErrorHandler(),
+      untilDestroyed(this),
+    ).subscribe((results) => {
+      this.poolEncryptionStatus = {};
+      results.forEach(({ poolName, encrypted }) => {
+        this.poolEncryptionStatus[poolName] = encrypted;
+      });
+      this.cdr.markForCheck();
     });
+  }
 
-    this.form.controls.destroy.valueChanges
-      .pipe(untilDestroyed(this))
-      .subscribe(() => this.resetNameInputValidState());
+  private determineDestinationPool(): void {
```

Comment:
How do you know this is the destination pool middleware is going to choose?

Reply:
It's the logic for multi pool provided in the mockup.

Reply:
Please check with backend team, dont think we can guess what the destination is going to be. We need their input.